### ISSUE TITLE
Data loss : When trying to add additional use cases and functionality in the prompt the old code files added are getting removed or lost

### ISSUE THREAD

**Issue by epicstudios3811** [stackblitz/bolt.new#1426]
### Describe the bug

When a file is being modified or updated the old existing code functions which were already there in the file are getting removed, Despite requiring changes in only a specific function or part of the code  here instead the entire file is being re-written fully having code loss were rest of the code that existed previously before update are also getting removed. This is a Showstopper. Please fix this ASAP. This is definitely limiting the bolt development experience.

### Link to the Bolt URL that caused the error

https://bolt.new/~/vitejs-vite-l2ec77

### Steps to reproduce

1) Just try adding new services under api oc convertApi or try fixing bug or update code file through command any other existing functinalities that existed already may get removed

### Expected behavior

Only the affected or functions that needs to be updated or rewritten should be updated not the entire file. Because of this issue unnecessarily I need to again waste my token to regenerate those lost code which again is not restored correctly when new use cases are extended.

### Platform

```
Browser name = Chrome
Full version = 130.0.0.0
Major version = 130
navigator.appName = Netscape
navigator.userAgent = Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36
performance.memory = {
  "totalJSHeapSize": 65761113,
  "usedJSHeapSize": 44144465,
  "jsHeapSizeLimit": 4294705152
}
Username = epicstudios3811
Chat ID = f0785717b928
```

**Comment by ranaprathap2** [comment]
Yes thanks for bringing it up, Occurs very often when trying to build and scale code definitely a showstopper. Needs attention.
